pkg/deployer/mock/actuator.go

```diff
@@ -74,7 +76,13 @@ func (a *actuator) Reconcile(ctx context.Context, req reconcile.Request) (reconc
 		return reconcile.Result{}, nil
 	}
 
-	if deployItem.Status.ObservedGeneration == deployItem.Generation {
+	err := dutils.HandleAnnotationsAndGeneration(ctx, a.c, a.log, deployItem)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if deployItem.Status.Phase != lsv1alpha1.ExecutionPhaseInit && deployItem.Status.Phase != lsv1alpha1.ExecutionPhaseDeleting {
```

Comment:
shouldn't the mock deployer be able to always reconcile?
we need this check in the container deployer as it otherweise would deploy multiple pods but the other controllers use the reconcile also for generic reconciles

Reply:
Actually, most deployers had some kind of check `deployitem.Status.ObservedGeneration == deployitem.Generation` there which I replaced with the new logic. Not sure about the mock deployer, but I can remove it here.